### PR TITLE
Metrics for how often we find existing quotes

### PR DESCRIPTION
**Pull request by fleupold**
I started looking into #1593 and noticed that our order quoting logic is quite complex. I wonder how much benefit there is in "fuzzy" matching order quotes by parameters, which seems to carry most of the branch complexity.

It would be substantially easier if we only exact matched quotes on a set of order parameters and recalculated quotes for all other matches.

This PR adds a metric to count how many quotes are matched by exact vs fuzzy match (and also logs if a quote is not found for what reason)

### CODE REVIEW

File: crates/shared/src/order_quoting.rs [hunk 1]
```diff
@@ -600,32 +600,39 @@ impl OrderQuoting for OrderQuoter {
...
                         .find(parameters, now, quote_kind)
                         .await?
-                        .ok_or(FindQuoteError::NotFound(None))?
+                        .ok_or(FindQuoteError::NotFound(None))
                 }
             };
-            Ok(Quote::new(Some(id), data))
+            match result {
+                Ok((found_id, data)) => {
+                    metrics::Metrics::found(id.map(|id| id == found_id).unwrap_or(false));
```
Comment (MartinquaXD) [inline crates/shared/src/order_quoting.rs:628]:
```suggestion
                    metrics::Metrics::found(id == Some(found_id));
```

File: crates/shared/src/order_quoting.rs [hunk 2]
```diff
@@ -702,6 +709,48 @@ fn quote_kind_from_signing_scheme(scheme: &QuoteSigningScheme) -> QuoteKind {
...
 }
 
+mod metrics {
+    use super::FindQuoteError;
+
+    #[derive(prometheus_metric_storage::MetricStorage)]
+    pub struct Metrics {
+        /// Counter for whether quotes where found by ID, fuzzy search or not
+        /// found at all and for what reason.
+        #[metric(name = "orderbook_quote_finding", labels("success", "type"))]
```
Comment (MartinquaXD) [inline crates/shared/src/order_quoting.rs:719]: I think the name is not needed here. Also metrics automatically get a prefix depending on the pod they get queried in so the `orderbook_` prefix is not needed.

**Review by MartinquaXD — APPROVED** [review]
While having metrics for an informed decision are nice the backend team is already pretty keen on dropping this fuzzy search to simplify the quoting process.

**Review by vkgnosis — APPROVED** [review]

**Comment by nlordell** [comment]
> It would be substantially easier if we only exact matched quotes on a set of order parameters and recalculated quotes for all other matches.

Agree. This was added for backwards compatibility with integrations that were not specifying quote IDs. If we remove this, we should be conscious of the fact that this may break existing integrations.

> This PR adds a metric to count how many quotes are matched by exact vs fuzzy match

What is the purpose of this metric? Is it just to determine if this feature is still being used or not? I _think_ you should be able to figure that out with Kibana without additional metrics. (will double check).

**Comment by nlordell** [comment]
Can this PR be closed?

**Pull request closed without merging**